lib/ddtrace/contrib/extensions.rb

```diff
@@ -1,21 +1,28 @@
 require 'set'
-require 'ddtrace/contrib/registry'
+require 'ddtrace/contrib'
+require 'ddtrace/configuration/settings'
 
 module Datadog
   module Contrib
     # Extensions that can be added to the base library
     # Adds registry, configuration access for integrations.
     module Extensions
       def self.extended(base)
-        Datadog.extend(Helpers)
-        Datadog.extend(Configuration)
-        Datadog::Configuration::Settings.include(Configuration::Settings)
+        base.extend(Helpers)
+        base.extend(Configuration)
       end
 
       # Helper methods for Datadog module.
       module Helpers
+        # Returns the global integration registry.
+        #
+        # This method is not safe to use while the tracer is initializing,
+        # thus access to the registry should go through
+        # ::Datadog::Contrib::REGISTRY for internal tracer work.
+        #
+        # External use of this method is always safe.
         def registry
-          configuration.registry
+          Contrib::REGISTRY
         end
       end
```

Comment:
I can take a stab at that and see how it looks

Reply:
After looking into it, I agree with this.
But I think this will modify too much to be able to fit nicely in this PR.

I documented this sentiment in code, so we know this is the direction we want for this part of our code in the future.

Reply:
Minor: I agree with avoiding the bigger change, but on my second go around reviewing this I was thinking that we may be able to just inline the `Helper` module into `datadog.rb`, and thus start a bit on the correct path.

Reply:
I tried to refactor `Helper`, but it looked very ugly without moving the rest of `extensions.rb` with it 😐

Reply:
Fair enough, thanks for making a try :)